### PR TITLE
update Singapore holidays

### PR DESCRIPTION
**Pull request by commenthol**
This PR builds on and updates #72 
- it uses fixed dates for Vesak day and Deepavali
- add substitute days if public holiday falls on a sunday
- adds tests

### CODE REVIEW

**Comment by commenthol** [comment]
Hi @ashlrem, Thanks for your initial PR #72. Could you please revise my changes and comment on those?

File: data/countries/SG.yaml [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+      1 Shawwal:
+        _name: 1 Shawwal
+        name:
+          en: Hari Raya Puasa
+      10 Dhu al-Hijjah:
+        _name: 10 Dhu al-Hijjah
+        name:
+          en: Hari Raya Haji
+      # https://en.wikipedia.org/wiki/Vesak
+      '2001-05-07':
```
Comment (gerryster) [inline data/countries/SG.yaml]: I like this pragmatic approach for now.

Comment (commenthol) [reply data/countries/SG.yaml]: well, it will work until we miss the yearly updates.

Comment (gerryster) [reply data/countries/SG.yaml]: Yea, I need to take the time to work through the calendar and determine the pattern.

**Comment by gerryster** [comment]
This looks good to me. @ashlrem: are you good with these changes?

**Comment by ashlrem** [comment]
> This looks good to me. @ashlrem: are you good with these changes?

Just checked. Yes. This is good to go. Apologies I wasn't able to get this done. I was doing my deliveries with work.

**Pull request merged**